angr/procedures/libc/strtol.py

```diff
@@ -140,33 +142,71 @@ def _char_to_val(char, base):
         """
         cases = []
         # 0-9
-        max_digit = claripy.BVV("9", 8)
-        min_digit = claripy.BVV("0", 8)
+        max_digit = state.se.BVV("9", 8)
+        min_digit = state.se.BVV("0", 8)
         if base < 10:
-            max_digit = claripy.BVV(chr(ord("0") + base), 8)
-        is_digit = claripy.And(char >= min_digit, char <= max_digit)
+            max_digit = state.se.BVV(chr(ord("0") + base), 8)
+        is_digit = state.se.And(char >= min_digit, char <= max_digit)
         # return early here so we don't add unnecessary statements
         if base <= 10:
             return is_digit, char - min_digit
 
         # handle alphabetic chars
-        max_char_lower = claripy.BVV(chr(ord("a") + base-10 - 1), 8)
-        max_char_upper = claripy.BVV(chr(ord("A") + base-10 - 1), 8)
-        min_char_lower = claripy.BVV(chr(ord("a")), 8)
-        min_char_upper = claripy.BVV(chr(ord("A")), 8)
+        max_char_lower = state.se.BVV(chr(ord("a") + base-10 - 1), 8)
+        max_char_upper = state.se.BVV(chr(ord("A") + base-10 - 1), 8)
+        min_char_lower = state.se.BVV(chr(ord("a")), 8)
+        min_char_upper = state.se.BVV(chr(ord("A")), 8)
 
         cases.append((is_digit, char - min_digit))
-        is_alpha_lower = claripy.And(char >= min_char_lower, char <= max_char_lower)
+        is_alpha_lower = state.se.And(char >= min_char_lower, char <= max_char_lower)
         cases.append((is_alpha_lower, char - min_char_lower + 10))
-        is_alpha_upper = claripy.And(char >= min_char_upper, char <= max_char_upper)
+        is_alpha_upper = state.se.And(char >= min_char_upper, char <= max_char_upper)
         cases.append((is_alpha_upper, char - min_char_upper + 10))
 
-        expression = claripy.Or(is_digit, is_alpha_lower, is_alpha_upper)
+        expression = state.se.Or(is_digit, is_alpha_lower, is_alpha_upper)
         # use the last case as the default, the expression will encode whether or not it's satisfiable
-        result = claripy.ite_cases(cases[:-1], cases[-1][1])
+        result = state.se.ite_cases(cases[:-1], cases[-1][1])
 
         return expression, result
-
+    @staticmethod
```

Comment:
I think both `_char_filter()` and `_digit_filter()` are for correcting the `strtol` implementation, which are irrelevant to `gets`. Can you please open a separate PR with only your fixes to `strtol`? We can merge them in first.